parachain/primitives/router/src/inbound/mod.rs

```diff
@@ -198,10 +224,17 @@ where
 					.encode()
 					.into(),
 			},
-			RefundSurplus,
-			// Clear the origin so that remaining assets in holding
-			// are claimable by the physical origin (BridgeHub)
-			ClearOrigin,
+			SetAppendix(Xcm(vec![
```

Comment:
So I am not sure what we gain from using SetAppendix at the end of an xcm message? SetAppendix works like the `finally` of a `try/catch/finally` block. Setting it at the end means it won't run if there are any errors in the instructions before it. The `Transact` always returns success, you have to call `ExpectTransactStatus` instruction to make it trigger xcm error handlers. So I would remove the SetAppendix completely and just add the DepositAsset after the ClearOrigin, or move the SetAppendix closer to the top so if it errors anywhere then the left over fees go to Bridgehub sovereign.

Reply:
Nice catch, I revert in

Reply:
On second thought I think we should rather move the SetAppendix after the BuyExecution but before the DepositAsset. That way any error after buying execution will result in the funds being returned to the bridgehub sovereign.

Reply:
https://github.com/Snowfork/snowbridge/pull/1096/commits/ec4ce325c1b7bf49d5ea530863375c9c6827ee8a